content/en/blog/low-hanging-performance/index.md

```diff
@@ -0,0 +1,460 @@
+
+```plantuml
+@startuml
+
+title Protocol State Machine
+
+state "Waiting Start Cmd/Msg" as wstart
+
+[*] -> wstart
+wstart --> PSM: InvitationCmd
+wstart --> PSM: ConnReqMsg
+
+state "ConnectionProtocolRunning" as PSM {
+
+	state "Sending ConnReq" as sreq
+	state "Waiting ConnResp" as wresp
+
+	'[*] -left-> sreq: InvitationCmd
+	[*] --> sreq: InvitationCmd
+
+	sreq --> wresp : http/200
+	sreq: do/send until http/200
+	
+	wresp: exit/return http 200
+
+	wresp --> [*]: ConnRespMsg
+
+||
+
+	state "Sending ConnResp" as sresp
+
+	[*] --> sresp: ConnReqMsg
+
+	sresp: do/send until http 200
+
+	sresp --> [*]: http/200
+}
+
+PSM --> [*]
+
+@enduml
+```
+
+Expressing things with the control-flow structures of imperative (or functional)
+programming languages is more challenging – especially when the correctness of
+the design should be verified. It seems that it’s easy to forget software
+quality attributes during fast-phasing programming if we use tools that only
+manage sequential text, i.e., code. At the code level, we should use functions
+that give us an abstraction hierarchy and help us to maintain readability.
+
+Moreover, since my studies of SW architecture’s quality attributes, I have
+understood that modifiability is more than modularity, re-usability, or using
+correct architectural styles like pipe and filter. Now we understand the importance of
+[TTD](https://en.wikipedia.org/wiki/Test-driven_development), [continuous
+deployment](https://en.wikipedia.org/wiki/Continuous_deployment),
+[DevOps](https://en.wikipedia.org/wiki/DevOps), etc. The practices above
+don't work only on *one* of the engineering domains. The best results are
+achieved **cross-domain engineering practices**. 
+
+In this post, I'll go through some fundamental Go tools and tricks you could use
+to achieve *decent execution speed of your (Go) software and maintain or
+increase its readability.*
+
+# Perfect Code?
+
+I suppose all of us programmers have heard of the infamous *premature optimization*:
+
+> *The real problem is that programmers have spent far too much time worrying
+> about efficiency in the wrong places and at the wrong times; premature
+> optimization is the root of all evil (or at least most of it) in programming.*
+
+
+That’s the full quote from the must-read *The Art of Computer Programming by
+Donald Knuth*. Like so many pearls of wisdom, they are a child of their own time.
+They are usually dangerously separated from their context to underline the
+message the next author wants to emphasize their statement. I believe most of us
+have read the quotation in its shortened format:
+
+> *Premature optimization is the root of all evil in programming.*
+
+Really? I don't think so.
+
+I claim that if **keeping performance is your second nature, it’ll not ruin the
+other quality attributes of your code, but the opposite**. All you need to do is
+to follow some basic rules with your muscle memory.
+
+## Performance Rules
```

Comment:
This summary is also good. Also, as a reader I would expect you to present these topics in this order, and I was a little surprised it didn't happen.